### PR TITLE
replace witness plugin with gradle dependency checksum verification

### PR DESCRIPTION
**Pull request by premek**
I noticed that a recent dependency change didn't needed checksum change. It was because the witness plugin only checked the listed checksums, not all the dependencies. If the dependency was not on the list, the new checksum was not verified.

Gradle now has its own verification mechanism, see https://docs.gradle.org/current/userguide/dependency_verification.html

After a dependency is added or a version is changed, the checksums could be regenerated using the following command.

```
gradle --write-verification-metadata sha256 help
````

 The help task is just used to discover as much as possible dependencies, and if subsequent builds fail with a verification error, you can re-execute generation with the appropriate tasks to "discover" more dependencies

 I verified that all the checksums from the removed file are present in the new one.

### CODE REVIEW

**Comment by premek** [comment]
I had to run it a few times in the CI to get all the mac and win dependencies right

File: build.gradle [hunk 1]
```diff
@@ -336,6 +335,7 @@ configure(project(':core')) {
             exclude(module: 'jackson-annotations')
             exclude(module: 'jackson-databind')
             exclude(module: 'bcprov-jdk15on')
+            exclude(group: 'org.slf4j', module: 'slf4j-simple')
```
Comment (premek) [inline build.gradle:338]: this is not related to the verification, it was just causing a warning in console when running

**Review by erciccione — COMMENTED** [review]
I checked out your PR locally and ran `./gradlew --write-verification-metadata sha256 help`. I get these changes after. Don't know if it's expected.

```
diff --git a/gradle/verification-metadata.xml b/gradle/verification-metadata.xml
index 9cd136a49ee..0ebb664303e 100644
--- a/gradle/verification-metadata.xml
          <artifact name="protoc-3.10.0-linux-x86_64.exe">
             <sha256 value="eed3ea189a99e3ad4e4209332e7161b255dc8f39bbde4c8e9fc25535f0f6f4f5" origin="Generated by Gradle"/>
          </artifact>
-      </component>
-      <component group="com.google.protobuf" name="protoc" version="3.10.0">
          <artifact name="protoc-3.10.0-osx-x86_64.exe">
             <sha256 value="a73d5bc8a2a4ab2a541901aa5ce81beacf8a7be5635be06a99c88f5ec10e840f" origin="Generated by Gradle"/>
          </artifact>
-      </component>
-      <component group="com.google.protobuf" name="protoc" version="3.10.0">
          <artifact name="protoc-3.10.0-windows-x86_64.exe">
             <sha256 value="996fe8523c4d6b70f6e390c9105fbe76023e8fc7aac0ad5e316ec0a48ad9a9b2" origin="Generated by Gradle"/>
          </artifact>
@@ -763,13 +759,9 @@
          <artifact name="protoc-gen-grpc-java-1.25.0-linux-x86_64.exe">
             <sha256 value="a712c0af4a0f7261a4880398072da6b482beca2307c2e5c42592d96841ca0ec2" origin="Generated by Gradle"/>
          </artifact>
-      </component>
-      <component group="io.grpc" name="protoc-gen-grpc-java" version="1.25.0">
          <artifact name="protoc-gen-grpc-java-1.25.0-osx-x86_64.exe">
             <sha256 value="83cc359bce8a615571c6fafc2979a143364b77a53f5b89a194aa8609c0c0c9fa" origin="Generated by Gradle"/>
          </artifact>
-      </component>
-       <component group="io.grpc" name="protoc-gen-grpc-java" version="1.25.0">
          <artifact name="protoc-gen-grpc-java-1.25.0-windows-x86_64.exe">
             <sha256 value="3186347362277c18082fac6de2525ca8311becbe29e95414fe55559b8dd2fbff" origin="Generated by Gradle"/>
          </artifact>

```

**Comment by premek** [comment]
That was caused by me manually adding them after running the build in windows and mac CI, I updated that.
It also needed rebasing and updating the monero-java checkusms

**Comment by woodser** [comment]
I'm getting this error in Eclipse after this change which I haven't been able to resolve:

<details>
<summary>The supplied phased action failed with an exception.</summary>
<br>
Dependency verification failed for org.projectlombok:lombok:1.18.12
88 artifacts failed verification:
  - annotations-13.0-sources.jar (org.jetbrains:annotations:13.0) from repository MavenRepo
  - annotations-4.1.1.4-sources.jar (com.google.android:annotations:4.1.1.4) from repository MavenRepo
  - aopalliance-1.0-sources.jar (aopalliance:aopalliance:1.0) from repository MavenRepo
  - apiguardian-api-1.1.0-sources.jar (org.apiguardian:apiguardian-api:1.1.0) from repository MavenRepo
  - bcpg-jdk15on-1.63-sources.jar (org.bouncycastle:bcpg-jdk15on:1.63) from repository MavenRepo
  - bcprov-jdk15on-1.63-sources.jar (org.bouncycastle:bcprov-jdk15on:1.63) from repository MavenRepo
  - bcprov-jdk15to18-1.63-sources.jar (org.bouncycastle:bcprov-jdk15to18:1.63) from repository MavenRepo
  - bitcoinj-2a80db4-sources.jar (com.github.bisq-network:bitcoinj:2a80db4) from repository maven
  - checker-qual-2.10.0-sources.jar (org.checkerframework:checker-qual:2.10.0) from repository MavenRepo
  - commons-codec-1.13-sources.jar (commons-codec:commons-codec:1.13) from repository MavenRepo
  - commons-compress-1.18-sources.jar (org.apache.commons:commons-compress:1.18) from repository MavenRepo
  - commons-io-2.6-sources.jar (commons-io:commons-io:2.6) from repository MavenRepo
  - commons-lang3-3.11-sources.jar (org.apache.commons:commons-lang3:3.11) from repository MavenRepo
  - commons-logging-1.2-sources.jar (commons-logging:commons-logging:1.2) from repository MavenRepo
  - core-2.0-sources.jar (com.google.zxing:core:2.0) from repository MavenRepo
  - easybind-1.0.3-sources.jar (org.fxmisc.easybind:easybind:1.0.3) from repository MavenRepo
  - error_prone_annotations-2.3.4-sources.jar (com.google.errorprone:error_prone_annotations:2.3.4) from repository MavenRepo
  - failureaccess-1.0.1-sources.jar (com.google.guava:failureaccess:1.0.1) from repository MavenRepo
  - fontawesomefx-8.0.0-sources.jar (de.jensd:fontawesomefx:8.0.0) from repository MavenRepo
  - fontawesomefx-commons-9.1.2-sources.jar (de.jensd:fontawesomefx-commons:9.1.2) from repository MavenRepo
  - fontawesomefx-materialdesignfont-2.0.26-9.1.2-sources.jar (de.jensd:fontawesomefx-materialdesignfont:2.0.26-9.1.2) from repository MavenRepo
  - grpc-api-1.25.0-sources.jar (io.grpc:grpc-api:1.25.0) from repository MavenRepo
  - grpc-context-1.25.0-sources.jar (io.grpc:grpc-context:1.25.0) from repository MavenRepo
  - grpc-core-1.25.0-sources.jar (io.grpc:grpc-core:1.25.0) from repository MavenRepo
  - grpc-netty-shaded-1.25.0-sources.jar (io.grpc:grpc-netty-shaded:1.25.0) from repository MavenRepo
  - grpc-protobuf-1.25.0-sources.jar (io.grpc:grpc-protobuf:1.25.0) from repository MavenRepo
  - grpc-protobuf-lite-1.25.0-sources.jar (io.grpc:grpc-protobuf-lite:1.25.0) from repository MavenRepo
  - grpc-stub-1.25.0-sources.jar (io.grpc:grpc-stub:1.25.0) from repository MavenRepo
  - gson-2.8.5-sources.jar (com.google.code.gson:gson:2.8.5) from repository MavenRepo
  - guava-28.2-jre-sources.jar (com.google.guava:guava:28.2-jre) from repository MavenRepo
  - guice-4.2.2-sources.jar (com.google.inject:guice:4.2.2) from repository MavenRepo
  - hamcrest-core-1.3-sources.jar (org.hamcrest:hamcrest-core:1.3) from repository MavenRepo
  - httpclient-4.5.12-sources.jar (org.apache.httpcomponents:httpclient:4.5.12) from repository MavenRepo
  - httpclient5-5.0-sources.jar (org.apache.httpcomponents.client5:httpclient5:5.0) from repository MavenRepo
  - httpcore-4.4.13-sources.jar (org.apache.httpcomponents:httpcore:4.4.13) from repository MavenRepo
  - httpcore5-5.0-sources.jar (org.apache.httpcomponents.core5:httpcore5:5.0) from repository MavenRepo
  - httpcore5-h2-5.0-sources.jar (org.apache.httpcomponents.core5:httpcore5-h2:5.0) from repository MavenRepo
  - j2objc-annotations-1.3-sources.jar (com.google.j2objc:j2objc-annotations:1.3) from repository MavenRepo
  - jackson-annotations-2.12.1-sources.jar (com.fasterxml.jackson.core:jackson-annotations:2.12.1) from repository MavenRepo
  - jackson-core-2.12.1-sources.jar (com.fasterxml.jackson.core:jackson-core:2.12.1) from repository MavenRepo
  - jackson-databind-2.12.1-sources.jar (com.fasterxml.jackson.core:jackson-databind:2.12.1) from repository MavenRepo
  - jackson-module-jaxb-annotations-2.12.1-sources.jar (com.fasterxml.jackson.module:jackson-module-jaxb-annotations:2.12.1) from repository MavenRepo
  - jakarta.activation-api-1.2.1-sources.jar (jakarta.activation:jakarta.activation-api:1.2.1) from repository MavenRepo
  - jakarta.xml.bind-api-2.3.2-sources.jar (jakarta.xml.bind:jakarta.xml.bind-api:2.3.2) from repository MavenRepo
  - javafx-base-15-sources.jar (org.openjfx:javafx-base:15) from repository MavenRepo
  - javafx-controls-15-sources.jar (org.openjfx:javafx-controls:15) from repository MavenRepo
  - javafx-fxml-15-sources.jar (org.openjfx:javafx-fxml:15) from repository MavenRepo
  - javafx-graphics-15-sources.jar (org.openjfx:javafx-graphics:15) from repository MavenRepo
  - javafx-swing-15-sources.jar (org.openjfx:javafx-swing:15) from repository MavenRepo
  - javase-2.0-sources.jar (com.google.zxing:javase:2.0) from repository MavenRepo
  - javax.annotation-api-1.2-sources.jar (javax.annotation:javax.annotation-api:1.2) from repository MavenRepo
  - javax.inject-1-sources.jar (javax.inject:javax.inject:1) from repository MavenRepo
  - jcip-annotations-1.0-sources.jar (net.jcip:jcip-annotations:1.0) from repository MavenRepo
  - jcsv-1.4.0-sources.jar (com.googlecode.jcsv:jcsv:1.4.0) from repository MavenRepo
  - jeromq-0.5.2-sources.jar (org.zeromq:jeromq:0.5.2) from repository MavenRepo
  - jfoenix-9.0.10-sources.jar (com.jfoenix:jfoenix:9.0.10) from repository MavenRepo
  - jnacl-1.0.0-sources.jar (eu.neilalexander:jnacl:1.0.0) from repository MavenRepo
  - jopt-simple-5.0.4-sources.jar (net.sf.jopt-simple:jopt-simple:5.0.4) from repository MavenRepo
  - jsonrpc4j-1.6.0.bisq.1-sources.jar (com.github.bisq-network:jsonrpc4j:1.6.0.bisq.1) from repository maven
  - jsr305-3.0.2-sources.jar (com.google.code.findbugs:jsr305:3.0.2) from repository MavenRepo
  - jtorctl-9b5ba2036b-sources.jar (com.github.JesusMcCloud:jtorctl:9b5ba2036b) from repository maven
  - junit-4.13.1-sources.jar (junit:junit:4.13.1) from repository MavenRepo
  - junit-jupiter-api-5.7.0-sources.jar (org.junit.jupiter:junit-jupiter-api:5.7.0) from repository MavenRepo
  - junit-jupiter-engine-5.7.0-sources.jar (org.junit.jupiter:junit-jupiter-engine:5.7.0) from repository MavenRepo
  - junit-jupiter-params-5.7.0-sources.jar (org.junit.jupiter:junit-jupiter-params:5.7.0) from repository MavenRepo
  - junit-platform-commons-1.7.0-sources.jar (org.junit.platform:junit-platform-commons:1.7.0) from repository MavenRepo
  - junit-platform-engine-1.7.0-sources.jar (org.junit.platform:junit-platform-engine:1.7.0) from repository MavenRepo
  - kotlin-logging-1.5.4-sources.jar (io.github.microutils:kotlin-logging:1.5.4) from repository MavenRepo
  - kotlin-stdlib-1.3.41-sources.jar (org.jetbrains.kotlin:kotlin-stdlib:1.3.41) from repository MavenRepo
  - kotlin-stdlib-common-1.3.41-sources.jar (org.jetbrains.kotlin:kotlin-stdlib-common:1.3.41) from repository MavenRepo
  - kotlin-stdlib-jdk7-1.3.41-sources.jar (org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.3.41) from repository MavenRepo
  - kotlin-stdlib-jdk8-1.3.41-sources.jar (org.jetbrains.kotlin:kotlin-stdlib-jdk8:1.3.41) from repository MavenRepo
  - logback-classic-1.1.11-sources.jar (ch.qos.logback:logback-classic:1.1.11) from repository MavenRepo
  - logback-core-1.1.11-sources.jar (ch.qos.logback:logback-core:1.1.11) from repository MavenRepo
  - lombok-1.18.12-sources.jar (org.projectlombok:lombok:1.18.12) from repository MavenRepo
  - monero-java-0.6.4-sources.jar (io.github.monero-ecosystem:monero-java:0.6.4) from repository MavenRepo
  - opencensus-api-0.21.0-sources.jar (io.opencensus:opencensus-api:0.21.0) from repository MavenRepo
  - opencensus-contrib-grpc-metrics-0.21.0-sources.jar (io.opencensus:opencensus-contrib-grpc-metrics:0.21.0) from repository MavenRepo
  - opentest4j-1.2.0-sources.jar (org.opentest4j:opentest4j:1.2.0) from repository MavenRepo
  - perfmark-api-0.19.0-sources.jar (io.perfmark:perfmark-api:0.19.0) from repository MavenRepo
  - proto-google-common-protos-1.12.0-sources.jar (com.google.api.grpc:proto-google-common-protos:1.12.0) from repository MavenRepo
  - protobuf-java-3.10.0-sources.jar (com.google.protobuf:protobuf-java:3.10.0) from repository MavenRepo
  - qrgen-1.3-sources.jar (net.glxn:qrgen:1.3) from repository MavenRepo
  - slf4j-api-1.7.30-sources.jar (org.slf4j:slf4j-api:1.7.30) from repository MavenRepo
  - tor-8db4a13-sources.jar (com.github.bisq-network.netlayer:tor:8db4a13) from repository maven
  - tor.external-8db4a13-sources.jar (com.github.bisq-network.netlayer:tor.external:8db4a13) from repository maven
  - tor.native-8db4a13-sources.jar (com.github.bisq-network.netlayer:tor.native:8db4a13) from repository maven
  - xz-1.6-sources.jar (org.tukaani:xz:1.6) from repository MavenRepo
If the artifacts are trustworthy, you will need to update the gradle/verification-metadata.xml file by following the instructions at https://docs.gradle.org/6.6.1/userguide/dependency_verification.html#sec:troubleshooting-verification
Open this report for more details: file:///Users/woodser/git/haveno/build/reports/dependency-verification/at-1644155057961/dependency-verification-report.html	line 0	haveno		Gradle Error Marker
</details>

**Comment by woodser** [comment]
Opened a PR which resolves the Eclipse error: https://github.com/premek/haveno/pull/3

**Comment by woodser** [comment]
Thanks for the fast turnaround.
